core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -289,11 +307,14 @@ private[spark] class Executor(
       var taskStart: Long = 0
       var taskStartCpu: Long = 0
       startGCTime = computeTotalGcTime()
+      var hasNonDefaultCpusPerTask = false
 
       try {
         val (taskFiles, taskJars, taskProps, taskBytes) =
           Task.deserializeWithDependencies(serializedTask)
 
+        hasNonDefaultCpusPerTask = taskProps.containsKey(TaskSchedulerImpl.CPUS_PER_TASK)
+        if (hasNonDefaultCpusPerTask) handleNonDefaultCpusPerTask(init = true)
```

Comment:
It could do more in future. More importantly this is Spark-side, so it does not need to know what extensions may want to do with this case -- maybe handle OOME, or maybe allocate more efficient thread-pools, or change thread priority etc.